ir/module.go

```diff
@@ -187,6 +188,160 @@ func (m *Module) String() string {
 	return buf.String()
 }
 
+type ioWriteWrapping struct {
+	size int
+	err  error
+}
+
+func (i *ioWriteWrapping) Fprintln(w io.Writer, a ...interface{}) {
+	if i.err != nil {
```

Comment:
Also, taking a closer look, I think that this check should be inverted, we only want to invoke further `fmt.Fprintln` calls if there has *not* yet been an error.

So either:

```diff
- 	if i.err != nil { 
+ 	if i.err == nil { 
```

Or simply an early return:

```go
if i.err != nil { 
   return
}
```

Reply:
I'm not pretty sure with this, for me, it means `if no error`

Reply:
Without inverting the check, the `fmt.Fprintln` call would never run, as `i.err` is initialized to `nil`. This the code inside the `if err != nil { }` block would never run.

Reply:
Oh, I got it, a typo bug.